packages/renderer/src/lib/preferences/PreferencesConnectionCreationOrEditRendering.svelte

```diff
@@ -35,63 +35,73 @@ import {
-export let callback: (
-  param: string,
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  data: any,
-  handlerKey: symbol,
-  collect: (key: symbol, eventName: 'log' | 'warn' | 'error' | 'finish', args: string[]) => void,
-  tokenId: number | undefined,
-  taskId: number | undefined,
-) => Promise<void>;
-export let taskId: number | undefined = undefined;
-export let disableEmptyScreen = false;
-export let hideCloseButton = false;
-export let connectionInfo: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo | undefined = undefined;
-export let inProgress = false;
-
-$: configurationValues = new Map<string, { modified: boolean; value: string | boolean | number }>();
-let operationStarted = false;
-let operationSuccessful = false;
-let operationCancelled = false;
-let operationFailed = false;
-export let pageIsLoading = true;
-let showLogs = false;
-let tokenId: number | undefined;
+interface Props {
+  properties: IConfigurationPropertyRecordedSchema[];
+  providerInfo: ProviderInfo;
+  propertyScope: string;
+  callback(
+    param: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    data: any,
+    handlerKey: symbol,
+    collect: (key: symbol, eventName: 'log' | 'warn' | 'error' | 'finish', args: string[]) => void,
+    tokenId: number | undefined,
+    taskId: number | undefined,
+  ): Promise<void>;
+  taskId?: number;
+  disableEmptyScreen?: boolean;
+  hideCloseButton?: boolean;
+  connectionInfo?: ProviderContainerConnectionInfo | ProviderKubernetesConnectionInfo;
+  inProgress?: boolean;
+  pageIsLoading?: boolean;
+}
 
+let {
+  properties = [],
+  providerInfo,
+  propertyScope,
+  callback,
+  taskId = undefined,
+  disableEmptyScreen = false,
+  hideCloseButton = false,
+  connectionInfo = undefined,
+  inProgress = $bindable(false),
+  pageIsLoading = true,
+}: Props = $props();
+
+let configurationValues = $state(new Map<string, { modified: boolean; value: string | boolean | number }>());
+let operationStarted = $state(false);
+let operationSuccessful = $state(false);
+let operationCancelled = $state(false);
+let operationFailed = false;
+let showLogs = $state(false);
+let tokenId: number | undefined = $state();
 let osMemory: string;
 let osCpu: string;
 let osFreeDisk: string;
-
 // get only ContainerProviderConnectionFactory scope fields that are starting by the provider id
-let configurationKeys: IConfigurationPropertyRecordedSchema[] = [];
-
-let isValid = true;
-let errorMessage: string | undefined = undefined;
-
-let formEl: HTMLFormElement;
-
+let configurationKeys: IConfigurationPropertyRecordedSchema[] = $state([]);
+let isValid = $state(true);
+let errorMessage: string | undefined = $state();
+let formEl: HTMLFormElement | undefined = $state();
 let globalContext: ContextUI;
-
-let connectionAuditResult: AuditResult | undefined;
-$: connectionAuditResult = undefined;
-
+let connectionAuditResult: AuditResult | undefined = $state();
```

Comment:
could you group the states together, and the non-state together too ?

Like

// states
const ...= states();

// non-reactive variables
...